addons/mail/static/src/models/thread_view/thread_view.js

```diff
@@ -103,6 +103,21 @@ registerModel({
             const { length, [length - 1]: lastMessageView } = this.messageViews;
             return lastMessageView ? replace(lastMessageView) : clear();
         },
+        /**
+         * @private
+         * @returns {FieldCommand}
+         */
+        _computeMessageListView() {
+            return (
+                (
+                    this.thread &&
+                    this.threadCache &&
+                    (this.threadCache.isLoaded || this.thread.isTemporary)
+                )
+                ? insertAndReplace()
+                : clear()
+            );
```

Comment:
Maybe like this? It makes more sense as top level "or" in this case, the "and" are only needed because we don't use nullish:
```suggestion
            return (
                (this.thread && this.thread.isTemporary) ||
                (this.threadCache && this.threadCache.isLoaded)
            ) ? insertAndReplace() : clear();
```